staging/src/k8s.io/apiserver/pkg/util/flowcontrol/request/list_work_estimator.go

```diff
@@ -82,23 +82,22 @@ func (e *listWorkEstimator) estimate(r *http.Request) WorkEstimate {
 		return WorkEstimate{Seats: maximumSeats}
 	}
 
-	// TODO: For resources that implement indexes at the watchcache level,
-	//  we need to adjust the cost accordingly
+	limit := numStored
+	if utilfeature.DefaultFeatureGate.Enabled(features.APIListChunking) && listOptions.Limit > 0 &&
+		listOptions.Limit < numStored {
+		limit = listOptions.Limit
+	}
+
 	var estimatedObjectsToBeProcessed int64
-	switch {
-	case isListFromCache:
-		// if we are here, count is known
-		estimatedObjectsToBeProcessed = count
-	default:
-		// Even if a selector is specified and we may need to list and go over more objects from etcd
-		// to produce the result of size <limit>, each individual chunk will be of size at most <limit>.
-		// As a result. the work estimate of the request should be computed based on <limit> and the actual
-		// cost of processing more elements will be hidden in the request processing latency.
-		estimatedObjectsToBeProcessed = listOptions.Limit
-		if estimatedObjectsToBeProcessed == 0 {
-			// limit has not been specified, fall back to count
-			estimatedObjectsToBeProcessed = count
-		}
+
+	if isListFromCache {
```

Comment:
nit: I think that using switch instead of multiple if else if.. branches is a bit clearer:
```
switch {
case isListFromCache:
  ...
case listOptions.FieldSelector != "" ...:
  ...
default:
  ...
}
```

Reply:
done